src/server.c

```diff
@@ -1672,8 +1673,21 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         if (listLength(server.clients_waiting_acks) && prev_fsynced_reploff != server.fsynced_reploff) dont_sleep = 1;
     }
 
-    /* Handle writes with pending output buffers. */
-    handleClientsWithPendingWritesUsingThreads();
+    /* Handle writes with pending output buffers.
+     * In the first phase, we will try to leverage io_uring to handle fsync asynchronously,
+     * just skip the client that has pending writes. */
+    handleClientsWithPendingWritesUsingThreads(1);
+
+    if (canFsyncUsingIOUring()) {
+        io_uring *io_uring = getIOUring();
+        /* Wait io_uring_prep_fsync finished. */
+        ioUringWaitFsyncBarrier(io_uring);
```

Comment:
I think we can stall here in `ioUringWaitFsyncBarrier`, especially on a single cpu machine or when all the CPUs are busy with the IO tasks. So I think we should port over the latency doctor logic here.